File: arb-www/src/schema/twitterapi/tweet.ts
export class TwitterSearchResponse {
  tweets: TwitterTweet[];
  has_next_page: boolean;
  next_cursor?: string;

  constructor(data: any) {
    this.tweets = data.tweets?.map((tweet: any) => new TwitterTweet(tweet)) || [];
    this.has_next_page = data.has_next_page || false;
    this.next_cursor = data.next_cursor;
  }

  static fromJSON(data: any): TwitterSearchResponse {
    return new TwitterSearchResponse(data);
  }
}

export class TwitterTweet {
  type: string;
  id: string;
  url: string;
  text: string;
  source: string;
  retweet_count: number;
  reply_count: number;
  like_count: number;
  quote_count: number;
  view_count: number;
  created_at: string;
  lang: string;
  bookmark_count: number;

  constructor(data: any) {
    this.type = data.type || '';
    this.id = data.id || '';
    this.url = data.url || '';
    this.text = data.text || '';
    this.source = data.source || '';
    this.retweet_count = data.retweetCount || 0;
    this.reply_count = data.replyCount || 0;
    this.like_count = data.likeCount || 0;
    this.quote_count = data.quoteCount || 0;
    this.view_count = data.viewCount || 0;
    this.created_at = data.createdAt || '';
    this.lang = data.lang || '';
    this.bookmark_count = data.bookmarkCount || 0;
  }
}

File: arb-rs/src/schema/twitterapi/tweet.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TwitterSearchResponse {
    pub tweets: Vec<TwitterTweet>,
    #[serde(rename = "has_next_page")]
    pub has_next_page: bool,
    #[serde(rename = "next_cursor")]
    pub next_cursor: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TwitterTweet {
    #[serde(rename = "type")]
    pub tweet_type: String,
    pub id: String,
    pub url: String,
    pub text: String,
    pub source: String,
    #[serde(rename = "retweetCount")]
    pub retweet_count: i64,
    #[serde(rename = "replyCount")]
    pub reply_count: i64,
    #[serde(rename = "likeCount")]
    pub like_count: i64,
    #[serde(rename = "quoteCount")]
    pub quote_count: i64,
    #[serde(rename = "viewCount")]
    pub view_count: i64,
    #[serde(rename = "createdAt")]
    pub created_at: String,
    pub lang: String,
    #[serde(rename = "bookmarkCount")]
    pub bookmark_count: i64,
    // Allow additional fields to be ignored
    #[serde(flatten)]
    pub additional_fields: serde_json::Value,
}
